src/rgw/rgw_lc_s3.cc

```diff
@@ -269,6 +352,14 @@ XMLObj *RGWLCXMLParser_S3::alloc_obj(const char *el)
     obj = new LCMPExpiration_S3();
   } else if (strcmp(el, "DaysAfterInitiation") == 0) {
     obj = new LCDays_S3();
+  } else if (strcmp(el, "StorageClass") == 0) {
```

Comment:
I think that while we're at it we should just switch everything to the new style. Would make code more readable and less prone to errors.

Reply:
Yes， I‘ve switched lifecycle xml decode to the new style.